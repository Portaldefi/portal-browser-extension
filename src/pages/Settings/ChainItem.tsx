import React, { SyntheticEvent, useEffect, useState } from 'react';
import { Checkbox, Grid, Header } from 'semantic-ui-react';

interface IIdentityItem {
  name: string,
  comment: string,
  checked: boolean,
  handleSetCheckState: any,
  index: number
};


export default ({ name, comment, checked, handleSetCheckState, index }: IIdentityItem) => {

  const [check, setCheck] = useState(true);

  useEffect(() => {
    setCheck(checked);
  }, []);

  const onChange = (e: SyntheticEvent, data: object) => {
    setCheck(!check);
    handleSetCheckState(index, !check);
  };

  return (
    <Grid className='p-16 py-9 bg-white'>
      <Grid.Column width='eleven' textAlign='left'>
        <Header as='p' className='description'>{name}</Header>
        <Header as='p' className='comment'>{comment}</Header>
      </Grid.Column>
      <Grid.Column width='five' verticalAlign='middle'>
        <Checkbox toggle checked={check} onChange={onChange} />
      </Grid.Column>
    </Grid>
  )
}