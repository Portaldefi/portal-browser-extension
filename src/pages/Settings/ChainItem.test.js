import ChainItem from './ChainItem';
import { create, act } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

describe('ChainItemComponent', () => {
  it('should work', async () => {
    let tree;
    act(() => {
      tree = create(
        <BrowserRouter>
          <ChainItem />
        </BrowserRouter>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});