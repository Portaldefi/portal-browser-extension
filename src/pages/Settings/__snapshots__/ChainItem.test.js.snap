// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainItemComponent should work 1`] = `
<div
  className="ui grid p-16 py-9 bg-white"
>
  <div
    className="left aligned eleven wide column"
  >
    <p
      className="ui header description"
    />
    <p
      className="ui header comment"
    />
  </div>
  <div
    className="middle aligned five wide column"
  >
    <div
      className="ui checked fitted toggle checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={true}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
</div>
`;
